File: C/Sully/Sully.c
#include <unistd.h>
#include <stdio.h>
#include <fcntl.h>
#include <stdlib.h>
int i=5;
char *s="#include <unistd.h>%c#include <stdio.h>%c#include <fcntl.h>%c#include <stdlib.h>%cint i=%d;%cchar *s=%c%s%c;%cint main(){if(!i)return 0;%cchar name[99]=%cSully_%%d.c%c;sprintf(name,name,i);int fd=open(name,O_CREAT|O_RDWR|O_TRUNC,0654);dprintf(fd,s,10,10,10,10,i,10,34,s,34,10,10,34,34,34,34,10,34,34,34,34,10);char cmd[99]=%cgcc -o Sully_%%d Sully_%%d.c%c;sprintf(cmd,cmd,i,i);close(fd);system(cmd);%cchar name1[99]=%cSully_%%d.c%c;sprintf(name1,name1,--i);fd=open(name1,O_CREAT|O_RDWR|O_TRUNC,0654);dprintf(fd,s,10,10,10,10,i,10,34,s,34,10,10,34,34,34,34,10,34,34,34,34,10);char cmd1[99]=%cgcc -o Sully_%%d Sully_%%d.c && ./Sully_%%d%c;sprintf(cmd1,cmd1,i,i,i);close(fd);system(cmd1);}%c";
int main(){if(!i)return 0;
char name[99]="Sully_%d.c";sprintf(name,name,i);int fd=open(name,O_CREAT|O_RDWR|O_TRUNC,0654);dprintf(fd,s,10,10,10,10,i,10,34,s,34,10,10,34,34,34,34,10,34,34,34,34,10);char cmd[99]="gcc -o Sully_%d Sully_%d.c";sprintf(cmd,cmd,i,i);close(fd);system(cmd);
char name1[99]="Sully_%d.c";sprintf(name1,name1,--i);fd=open(name1,O_CREAT|O_RDWR|O_TRUNC,0654);dprintf(fd,s,10,10,10,10,i,10,34,s,34,10,10,34,34,34,34,10,34,34,34,34,10);char cmd1[99]="gcc -o Sully_%d Sully_%d.c && ./Sully_%d";sprintf(cmd1,cmd1,i,i,i);close(fd);system(cmd1);}

File: Go/Sully/Sully.go
package main
import (
"fmt"
"os"
"os/exec"
)
var i=5;
var s ="package main%cimport (%c%cfmt%c%c%cos%c%c%cos/exec%c%c)%cvar i=%d;%cvar s =%c%s%c;%cfunc compile(name,output string){var cmd=exec.Command(%cgo%c,%cbuild%c,%c-o%c,output,name);cmd.Run();}%cfunc run(bin string){exec.Command(bin).Run();}%cfunc write(filename string){var file,_ = os.OpenFile(filename,os.O_RDWR|os.O_CREATE,0755);file.WriteString(fmt.Sprintf(s,10,10,34,34,10,34,34,10,34,34,10,10,i,10,34,s,34,10,34,34,34,34,34,34,10,10,10,34,34,34,34,34,34,34,34,34,34,34,34,34,34,10));file.Close();};%cfunc main(){if(i!=0){write(fmt.Sprintf(%cSully_%%d.go%c,i));compile(fmt.Sprintf(%cSully_%%d.go%c,i),fmt.Sprintf(%cSully_%%d%c,i));i--;write(fmt.Sprintf(%cSully_%%d.go%c,i));compile(fmt.Sprintf(%cSully_%%d.go%c,i),fmt.Sprintf(%cSully_%%d%c,i));run(fmt.Sprintf(%c./Sully_%%d%c,i));}}%c";
func compile(name,output string){var cmd=exec.Command("go","build","-o",output,name);cmd.Run();}
func run(bin string){exec.Command(bin).Run();}
func write(filename string){var file,_ = os.OpenFile(filename,os.O_RDWR|os.O_CREATE,0755);file.WriteString(fmt.Sprintf(s,10,10,34,34,10,34,34,10,34,34,10,10,i,10,34,s,34,10,34,34,34,34,34,34,10,10,10,34,34,34,34,34,34,34,34,34,34,34,34,34,34,10));file.Close();};
func main(){if(i!=0){write(fmt.Sprintf("Sully_%d.go",i));compile(fmt.Sprintf("Sully_%d.go",i),fmt.Sprintf("Sully_%d",i));i--;write(fmt.Sprintf("Sully_%d.go",i));compile(fmt.Sprintf("Sully_%d.go",i),fmt.Sprintf("Sully_%d",i));run(fmt.Sprintf("./Sully_%d",i));}}
